sklearn/metrics/_classification.py

```diff
@@ -3134,13 +3134,16 @@ def hinge_loss(y_true, pred_decision, *, labels=None, sample_weight=None):
 @validate_params(
     {
         "y_true": ["array-like"],
-        "y_prob": ["array-like"],
+        "y_proba": ["array-like", None],
         "sample_weight": ["array-like", None],
         "pos_label": [Real, str, "boolean", None],
+        "y_prob": ["array-like", StrOptions({"deprecated"})],
```

Comment:
`None` and `"deprecated"` both need to be a hidden option.

Reply:
Done